src/bb/image-io/rt_u3v.h

```diff
@@ -312,224 +286,93 @@ class U3V {
-                    throw ::std::runtime_error("Buffer is null");
-                }
-                devices_[i].frame_count_ += 1;
-                memcpy(outs[i].data(), arv_buffer_get_part_data(bufs[i], 0, nullptr), size);
-            }
-    }
-
-    void get(std::vector<Halide::Buffer<>>& outs) {
-
-        if (sim_mode_){
-            get_buf_sim(outs);
-            return;
-        }
-
-        auto timeout_us = 30 * 1000 * 1000;
-
-        int32_t num_device = num_sensor_;
-        std::vector<ArvBuffer *> bufs(num_device);
-
-        // default is OperationMode::Came1USB1
-        if (operation_mode_ == OperationMode::Came2USB2 || operation_mode_ == OperationMode::Came1USB1){
-
-            // if aravis output queue length is more than N (where N > 1) for all devices, pop all N-1 buffer
-            if (realtime_display_mode_){
-                std::vector<int32_t> N_output_buffers(num_device);
-                for (auto i = 0; i < num_device; ++i){
-                    int32_t num_input_buffer;
-                    arv_stream_get_n_buffers(devices_[i].stream_, &num_input_buffer, &(N_output_buffers[i]));
-                }
-                // if all stream has N output buffers, discard N-1 of them
-                for(auto i = 0; i < num_device; ++i){
-                    for (auto j = 0; j < N_output_buffers[i]-1; ++j){
-                        bufs[i] = arv_stream_timeout_pop_buffer (devices_[i].stream_, timeout_us);
-                        if (bufs[i] == nullptr){
-                            log::error("pop_buffer(L2) failed due to timeout ({}s)", timeout_us*1e-6f);
-                            throw ::std::runtime_error("buffer is null");
-                        }
-                        devices_[i].frame_count_ = is_gendc_
-                            ? static_cast<uint32_t>(get_frame_count_from_genDC_descriptor(bufs[i], devices_[i]))
-                            : static_cast<uint32_t>(arv_buffer_get_timestamp(bufs[i]) & 0x00000000FFFFFFFF);
-                        i == 0 ?
-                            log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20}) [skipped for realtime display]", devices_[i].frame_count_, "") :
-                            log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20}) [skipped for realtime display]", "", devices_[i].frame_count_);
-                        arv_stream_push_buffer(devices_[i].stream_, bufs[i]);
-
-                    }
-                }
-
-            }
-
-            // get the first buffer for each stream
-            for (auto i = 0; i< devices_.size(); ++i) {
-                bufs[i] = arv_stream_timeout_pop_buffer (devices_[i].stream_, timeout_us);
-                if (bufs[i] == nullptr){
-                    log::error("pop_buffer(L1) failed due to timeout ({}s)", timeout_us*1e-6f);
-                    throw ::std::runtime_error("Buffer is null");
+     void setFrameSync(std::vector<ArvBuffer *> &bufs, int timeout_us){
```

Comment:
It sounds better to be:

- Following consistent naming convention (snake case)
- More proper name to indicate what is done by this function.  Is it setting "frame sync"?